ouroboros-network-protocols/test-cddl/Main.hs

```diff
@@ -590,6 +637,61 @@ prop_encodePeerSharing
 prop_encodePeerSharing spec =
   validateEncoder spec (codecPeerSharing encodeRemoteAddress decodeRemoteAddress)
 
+newtype NtNVersionV7To10    = NtNVersionV7To10 NodeToNodeVersion
+  deriving Show
+newtype NtNVersionV11       = NtNVersionV11 NodeToNodeVersion
+  deriving Show
+newtype NtNVersionV11ToLast = NtNVersionV11ToLast NodeToNodeVersion
+  deriving Show
+
+instance Arbitrary NtNVersionV7To10 where
+  arbitrary = NtNVersionV7To10 <$> elements [NodeToNodeV_7 .. NodeToNodeV_10]
+
+instance Arbitrary NtNVersionV11ToLast where
+  arbitrary = NtNVersionV11ToLast <$> elements [NodeToNodeV_11 ..]
+
+instance Arbitrary NodeToNodeVersionData where
+    arbitrary =
+      NodeToNodeVersionData
+        <$> (NetworkMagic <$> arbitrary)
+        <*> oneof [ pure InitiatorOnlyDiffusionMode
+                  , pure InitiatorAndResponderDiffusionMode
+                  ]
+        <*> oneof [ pure NoPeerSharing
+                  , pure PeerSharingPrivate
+                  , pure PeerSharingPublic
+                  ]
+        <*> arbitrary
+
+newtype NtNVersionDataV11ToLast = NtNVersionDataV11ToLast (NodeToNodeVersion, NodeToNodeVersionData)
+  deriving Show
+
+instance Arbitrary NtNVersionDataV11ToLast where
+  arbitrary = do
+    NtNVersionV11ToLast ntnVersion <- arbitrary
+    ntnVersionData <- arbitrary
+    let ntnVersionData' = case ntnVersion of
+          NodeToNodeV_11 -> ntnVersionData { NtNVersion.query = False }
```

Comment:
Why do we set the query flag to `False` in `NodeToNodeV_11`?  It was added in `NodeToNodeV_11`.

Reply:
I thought it got added in V_12, V_11 was PeerSharing, right?

Reply:
nvm, you are right